homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,428 @@
+                )
+                colortemp = kelvin
+                _LOGGER.debug(
+                    "[wizlight %s] kelvin changed and send to bulb: %s",
+                    self._light.ip,
+                    colortemp,
+                )
+
+            sceneid = None
+            if ATTR_EFFECT in kwargs:
+                sceneid = self._light.get_id_from_scene_name(kwargs[ATTR_EFFECT])
+
+            if sceneid == 1000:  # rhythm
+                pilot = PilotBuilder()
+            else:
+                pilot = PilotBuilder(
+                    brightness=brightness, colortemp=colortemp, scene=sceneid
+                )
+                _LOGGER.debug(
+                    "[wizlight %s] Pilot will be send with brightness=%s, colortemp=%s, scene=%s",
+                    self._light.ip,
+                    brightness,
+                    colortemp,
+                    sceneid,
+                )
+
+            sceneid = None
+            if ATTR_EFFECT in kwargs:
+                sceneid = self._light.get_id_from_scene_name(kwargs[ATTR_EFFECT])
+
+            if sceneid == 1000:  # rhythm
+                pilot = PilotBuilder()
+            else:
+                pilot = PilotBuilder(
+                    brightness=brightness, colortemp=colortemp, scene=sceneid
+                )
+
+        await self._light.turn_on(pilot)
+
+    async def async_turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        await self._light.turn_off()
+
+    @property
+    def color_temp(self):
+        """Return the CT color value in mireds."""
+        return self._temperature
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        if self._bulbtype is None:
+            return color_utils.color_temperature_kelvin_to_mired(6500)
+
+        try:
+            return color_utils.color_temperature_kelvin_to_mired(
+                self._bulbtype.kelvin_range.max
+            )
+
+        except WizLightNotKnownBulb:
+            _LOGGER.info("Kelvin is not present in the library. Fallback to 6500")
+            return color_utils.color_temperature_kelvin_to_mired(6500)
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        if self._bulbtype is None:
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+        try:
+            return color_utils.color_temperature_kelvin_to_mired(
+                self._bulbtype.kelvin_range.min
+            )
+        except WizLightNotKnownBulb:
+            _LOGGER.info("Kelvin is not present in the library. Fallback to 2200")
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        if self._bulbtype:
+            return self.featuremap()
+        # fallback
+        return SUPPORT_FEATURES_RGB
```

Comment:
Ideally these get set as `self._attr_min_mireds`, `self._attr_max_mireds`, and `self._attr_supported_features` so they don't have to get built every time the state updates